File: src/main/java/io/lsdconsulting/lsd/distributed/access/repository/InterceptedDocumentRepository.java
package io.lsdconsulting.lsd.distributed.access.repository;

import io.lsdconsulting.lsd.distributed.access.model.InterceptedInteraction;

import java.util.List;

public interface InterceptedDocumentRepository {
    void save(final InterceptedInteraction interceptedInteraction);
    List<InterceptedInteraction> findByTraceIds(final String... traceId);
}

File: src/main/kotlin/io/lsdconsulting/lsd/distributed/access/repository/InterceptedDocumentRepository.kt
package io.lsdconsulting.lsd.distributed.access.repository

import io.lsdconsulting.lsd.distributed.access.model.InterceptedInteraction

interface InterceptedDocumentRepository {
    fun save(interceptedInteraction: InterceptedInteraction)
    fun findByTraceIds(vararg traceId: String): List<InterceptedInteraction>
    fun isActive(): Boolean
}
